index.js

```diff
@@ -483,20 +483,20 @@ function get(map, key, def) {
 }
 
 function callMiddlewareFunction(fn, context, args, next) {
-  let maybePromise;
+  let maybePromiseLike;
   try {
-    maybePromise = fn.apply(context, args);
+    maybePromiseLike = fn.apply(context, args);
   } catch (error) {
     return next(error);
   }
 
-  if (isPromise(maybePromise)) {
-    maybePromise.then(() => next(), err => next(err));
+  if (isPromiseLike(maybePromiseLike)) {
+    maybePromiseLike.then(() => next(), err => next(err));
   }
 }
 
-function isPromise(v) {
-  return v != null && typeof v.then === 'function';
+function isPromiseLike(v) {
+  return v instanceof Promise || (typeof v === 'object' && v !== null && typeof v.then === 'function');
```

Comment:
Is there a reason for the `instanceof Promise` check? The `.then === 'function'` check will catch promises, so that isn't necessary